include/envoy/network/connection_handler.h

```diff
@@ -43,13 +43,29 @@ class ConnectionHandler {
    */
   virtual void addListener(ListenerConfig& config) PURE;
 
+  /**
+   * Adds a listener to the handler and replace the existing listener.
+   * @param overrided_listener supplies the tag of the existing listener.
+   * @param config listener configuration options.
+   */
+  virtual void addIntelligentListener(uint64_t overrided_listener, ListenerConfig& config) PURE;
```

Comment:
What is meant by an "intelligent" listener? Aren't all listeners going to be intelligent? Can we just merge this behavior into the existing interface?

Reply:
Conceptually in the particular path, the new listener would take over the `existing` ActiveListener at worker, instead of destroy new after create.
The destroy-old-after-create-new is required in below cases:
1. fresh new listener, no such old instance
2. active listener cannot support it *yet*: udp-listener, huge changing(including listener filter chain update) where I don't have a strong confidence to manipulate the life cycles of the dependencies.

Despite these cases, I can merge the 2 flows together into single function. As long as you feel it's easier to review.

Reply:
Yes IMO we should just be modifying the existing listener. I think it's fine if we start with very simple cases in that path, so let's do that.